wuvt/admin/views.py

```diff
@@ -492,6 +492,29 @@ def donation_index():
         if 'reset_stats' in request.form:
             redis_conn.set('donation_stats_start',
                            datetime.datetime.utcnow().isoformat())
+        if 'set_paid' in request.form:
+            # TODO log who set as paid
+            o = Order.query.get(request.form["id"])
+            o.set_paid(method="check")
+            db.session.add(o)
+            try:
+                db.session.commit()
+            except:
+                db.session.rollback()
+                raise
+
+        # TODO support setting asynchronously, rather than redirecting
+        if 'set_shipped' in request.form:
+            # TODO log who set as paid
+            o = Order.query.get(request.form["id"])
+            o.set_shipped()
+            db.session.add(o)
```

Comment:
What is being added here? This call doesn't seem correct; removing it should work.

Reply:
is my memory of sqlalchemy way off? I thought any db modification required you add it to the session? Or is it implicity added when you read from the DB?

Reply:
nevermind. I read the docs. indeed.